src/Illuminate/Container/Container.php

```diff
@@ -940,6 +940,16 @@ public function getBindings()
     }
 
     /**
+     * Return the types which are bound in this container, except for aliases.
+     *
+     * @return array
+     */
+    public function getAbstractTypes()
+    {
+        return array_unique(array_merge(array_keys($this->bindings), array_keys($this->instances)));
```

Comment:
It should also be noted that array_unique has quartic time complexity, so should be avoided.